modifications For nim-0.16

encorehu opened the pull request:
fix Warning: expr is deprecated [Deprecated]
fix Error: type mismatch: got (typedesc[Model])
etc.

for nim-lang 0.16

there are no tests, please check more carefully.

encorehu committed:
Warning: expr is deprecated [Deprecated]

encorehu committed:
Warning: expr is deprecated [Deprecated]

encorehu committed:
Warning: expr is deprecated [Deprecated]

encorehu committed:
Warning: expr is deprecated [Deprecated]

encorehu committed:
Warning: expr is deprecated [Deprecated]

encorehu committed:
fix Error: type mismatch: got (typedesc[Model])

encorehu committed:
fix Error: type mismatch: got (typedesc[Model])

encorehu committed:
fix Error: type mismatch: got (typedesc[Model])

encorehu committed:
debug

encorehu committed:
debug

encorehu committed:
fix Error: node has no type

encorehu committed:
debug

encorehu committed:
debug

encorehu committed:
fix result is @[] empty, recList is not objectTy[1], but is objectTy[2]

encorehu committed:
SIGSEGV: Illegal storage access. (Attempt to read from nil?)

encorehu committed:
fix Error: type mismatch: got (ref User, string)

but expected one of:

macro fieldIndex(sym: Model; field: untyped): untyped

encorehu committed:
#getType[2] is nnkRecList

encorehu committed:
Term rewriting fix Error: type mismatch: got (typedesc[string], ref User, int literal(0))

but expected one of:

template loadField[](T: typedesc[string]; user: Model; field: int32): string

template loadField[](T: typedesc[int]; user: Model; field: int32): int

template loadField[](T: typedesc[float]; user: Model; field: int32): float

template loadField[](T: typedesc[bool]; user: Model; field: int32): bool

encorehu committed:
use same scheme, ref Model

encorehu committed:
##NOT HAPPENED YET. but I changed objectTy[0] to objectTy[1]

encorehu committed:
add comment ##NOT HAPPENED YET. but I changed objectTy[0] to objectTy[1]

encorehu committed:
getType is correct form

Kentoseth commented:
This PR fails with nim 0.17.3:

`nim c expr2sql.nim`

```
expr2sql.nim(27, 41) template/generic instantiation from here
/.../nim-orm/orm.nim(194, 30) template/generic instantiation from here
lib/system.nim(1169, 69) Error: type mismatch: got (set[int8], int32)
but expected one of: 
proc contains[T](s: Slice[T]; value: T): bool
proc contains[T](x: set[T]; y: T): bool
proc contains[T](a: openArray[T]; item: T): bool
proc contains(s, sub: string): bool
proc contains(s: string; chars: set[char]): bool
proc contains(s: string; c: char): bool
```

encorehu committed:
fix Error: type mismatch: got <set[int8], int32>

change int32 to int8. or you will need to cast objectTyFieldIndex() int32 to int8

encorehu committed:
add comments: int8 can't be modified as int32, Error: set is too large

encorehu committed:
add comments

encorehu committed:
add comments

encorehu committed:
add comments

encorehu committed:
fix Warning: $ is deprecated and fix Warning: ident is deprecated

encorehu committed:
add comments of $n[0] == n[0].strVal

encorehu committed:
add comments

encorehu committed:
change "not implemented", to "Not Implemented Yet"

encorehu committed:
add macro items, used for iterator, and add usage of for x in Model

encorehu committed:
add doc of for user in User:

encorehu committed:
doc add macro items

nanoant commented:
Dear @encorehu, I am currently on business trip, but I wanted to know that I am alive and I received PR notification. Anyways, I will try to review your changes ASAP.

encorehu committed:
add json tests for nim-orm, Error: undeclared field: 'loaded'
